File: app/src/main/java/com/happs/ximand/ringcontrol/model/mapper/impl/CursorToTrustedDeviceMapper.java
package com.happs.ximand.ringcontrol.model.mapper.impl;

import android.database.Cursor;

import com.happs.ximand.ringcontrol.model.mapper.Mapper;
import com.happs.ximand.ringcontrol.model.object.TrustedDevice;

import static com.happs.ximand.ringcontrol.model.database.TrustedDevicesDatabaseHelper.DB_FIELD_DATE;
import static com.happs.ximand.ringcontrol.model.database.TrustedDevicesDatabaseHelper.DB_FIELD_ID;
import static com.happs.ximand.ringcontrol.model.database.TrustedDevicesDatabaseHelper.DB_FIELD_MANUFACTURER;
import static com.happs.ximand.ringcontrol.model.database.TrustedDevicesDatabaseHelper.DB_FIELD_MODEL;

public class CursorToTrustedDeviceMapper implements Mapper<Cursor, TrustedDevice> {

    @Override
    public TrustedDevice map(Cursor from) {
        int idColumnIndex = from.getColumnIndex(DB_FIELD_ID);
        int manufacturerColumnIndex = from.getColumnIndex(DB_FIELD_MANUFACTURER);
        int modelColumnIndex = from.getColumnIndex(DB_FIELD_MODEL);
        int dateColumnIndex = from.getColumnIndex(DB_FIELD_DATE);

        int id = from.getInt(idColumnIndex);
        String manufacturer = from.getString(manufacturerColumnIndex);
        String model = from.getString(modelColumnIndex);
        String date = from.getString(dateColumnIndex);

        return new TrustedDevice(id, manufacturer, model, date);
    }

}

File: app/src/main/java/com/happs/ximand/ringcontrol/model/mapper/impl/CursorToTrustedDeviceMapper.kt
package com.happs.ximand.ringcontrol.model.mapper.impl

import android.database.Cursor
import com.happs.ximand.ringcontrol.model.`object`.TrustedDevice
import com.happs.ximand.ringcontrol.model.database.TrustedDevicesDatabaseHelper
import com.happs.ximand.ringcontrol.model.mapper.Mapper

class CursorToTrustedDeviceMapper : Mapper<Cursor, TrustedDevice> {
    override fun map(from: Cursor): TrustedDevice {
        val idColumnIndex = from.getColumnIndex(TrustedDevicesDatabaseHelper.DB_FIELD_ID)
        val manufacturerColumnIndex = from.getColumnIndex(TrustedDevicesDatabaseHelper.DB_FIELD_MANUFACTURER)
        val modelColumnIndex = from.getColumnIndex(TrustedDevicesDatabaseHelper.DB_FIELD_MODEL)
        val dateColumnIndex = from.getColumnIndex(TrustedDevicesDatabaseHelper.DB_FIELD_DATE)
        val id = from.getInt(idColumnIndex)
        val manufacturer = from.getString(manufacturerColumnIndex)
        val model = from.getString(modelColumnIndex)
        val date = from.getString(dateColumnIndex)
        return TrustedDevice(id, manufacturer, model, date)
    }
}